.github/workflows/eks-ci.yml

```diff
@@ -46,8 +42,107 @@ jobs:
       - id: set-matrix
         run: echo "matrix=${{ env.TEST_MATRIX}}" >> $GITHUB_OUTPUT
 
+  Build:
+    name: Build
+    runs-on: ubuntu-20.04
+    timeout-minutes: 15
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v2
+
+      - name: Init Hermit
+        run: ./bin/hermit env -r >> $GITHUB_ENV
+
+      - name: cache go dependencies
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/go/pkg/mod
+            ~/.cache/go-build
+          key: ${{ runner.os }}-go-${{ matrix.go-version }}-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-${{ matrix.go-version }}-
+
+      - name: build cloudbeat binary
```

Comment:
Since the original plan is to merge eks-ci workflow with the pre-merge workflow (Stage 5 in EKS-CI epic), refactoring and finding optimal solution is less relevant in this case.

Reply:
got it :+1: